src/app/cli/CommandLineInterface.java

```diff
@@ -63,15 +92,280 @@ public void launch(){
+                        continue;
+                    }
+                    for(int i = 0; i < collections.length; ++i){
+                        for(int j = i+1; j < collections.length; ++j){
+                            Collection first = collections[i];
+                            Collection second = collections[j];
+                            if(first.coll_name().compareTo(second.coll_name()) > 0) {
+                                collections[i] = second;
+                                collections[j] = first;
+                            }
+                        }
+                    }
+                    for (Collection collection : collections) {
+                        System.out.println("Name:\t" + collection.coll_name());
+                        System.out.println("# of games\t" + collection.games().length);
+                        System.out.println("Play time:\t" + app.total_playtime_collection(collection));
+                    }
+                    System.out.println();
+                }
+
+                case COLLECTION_ADD -> {
+                    Collection[] collections_list_init = app.get_collections();
+                    ArrayList<Collection> collections_list = new ArrayList<>();
+                    for (Collection collection : collections_list_init) {
+                        if (collection.coll_name().equals(tokens[1])) {
+                            collections_list.add(collection);
+                        }
+                    }
+                    if(collections_list.size() == 0) {
+                        System.out.println("You do not have a collection with that name.");
+                        continue;
+                    }
+                    Collection selected_coll = collections_list.get(0);
+
+                    if(collections_list.size() > 1) {
+                        System.out.println("Which collection would you like to add the game to (enter the number)?");
+                        for(int i = 0; i < collections_list.size(); ++i) {
+                            Collection curr_coll = collections_list.get(i);
+                            System.out.println(i+1 + ".\tName: " + curr_coll.coll_name() + "\n\t# of games: " +
+                                    curr_coll.games().length + "\n\tPlay time: " +
+                                    app.total_playtime_collection(curr_coll) + "\n");
+                        }
+                        input = in.nextLine();
+                        int input_to_int = Integer.parseInt(input);
+                        selected_coll = collections_list.get(input_to_int-1);
+                    }
+                    Game[] game_list = app.search_game_name(tokens[2]);
+
+                    if(game_list.length == 0) {
+                        System.out.println("There is not a game with that name.");
+                        continue;
+                    }
+
+                    Game selected_game = game_list[0];
```

Comment:
shouldn't be initialized, or maybe initialized to `null`. ig it doesn't really matter but still

Reply:
I did it like that because it carries through, if the if statement doesn't apply you already have the value

Reply:
oh i see

other than that it should validate input and imo reprompt until they input a valid number (should be a method for this)

Reply:
Agree, will put that in